tests/src/Interop/Variant/VariantNative.cpp

```diff
@@ -0,0 +1,349 @@
+    {
+        printf("Invalid format. Expected VT_UI1.\n");
+        return FALSE;
+    }
+
+    return value.cVal == NumericValue ? TRUE : FALSE;
+}
+
+extern "C" BOOL DLL_EXPORT STDMETHODCALLTYPE Marshal_ByValue_SByte(VARIANT value)
+{
+    if (value.vt != VT_I1)
+    {
+        printf("Invalid format. Expected VT_I1.\n");
+        return FALSE;
+    }
+
+    return value.bVal == NumericValue ? TRUE : FALSE;
+}
+
+extern "C" BOOL DLL_EXPORT STDMETHODCALLTYPE Marshal_ByValue_Int16(VARIANT value)
+{
+    if (value.vt != VT_I2)
+    {
+        printf("Invalid format. Expected VT_I2.\n");
+        return FALSE;
+    }
+
+    return value.iVal == NumericValue ? TRUE : FALSE;
+}
+
+extern "C" BOOL DLL_EXPORT STDMETHODCALLTYPE Marshal_ByValue_UInt16(VARIANT value)
+{
+    if (value.vt != VT_UI2)
+    {
+        printf("Invalid format. Expected VT_UI2.\n");
+        return FALSE;
+    }
+
+    return value.uiVal == NumericValue ? TRUE : FALSE;
+}
+extern "C" BOOL DLL_EXPORT STDMETHODCALLTYPE Marshal_ByValue_Int32(VARIANT value)
+{
+    if (value.vt != VT_I4)
+    {
+        printf("Invalid format. Expected VT_I4.\n");
+        return FALSE;
+    }
+
+    return value.lVal == NumericValue ? TRUE : FALSE;
+}
+
+extern "C" BOOL DLL_EXPORT STDMETHODCALLTYPE Marshal_ByValue_UInt32(VARIANT value)
+{
+    if (value.vt != VT_UI4)
+    {
+        printf("Invalid format. Expected VT_UI4.\n");
+        return FALSE;
+    }
+
+    return value.ulVal == NumericValue ? TRUE : FALSE;
+}
+
+extern "C" BOOL DLL_EXPORT STDMETHODCALLTYPE Marshal_ByValue_Int64(VARIANT value)
+{
+    if (value.vt != VT_I8)
+    {
+        printf("Invalid format. Expected VT_I8.\n");
+        return FALSE;
+    }
+
+    return value.llVal == NumericValue ? TRUE : FALSE;
+}
+
+extern "C" BOOL DLL_EXPORT STDMETHODCALLTYPE Marshal_ByValue_UInt64(VARIANT value)
+{
+    if (value.vt != VT_UI8)
+    {
+        printf("Invalid format. Expected VT_UI8.\n");
+        return FALSE;
+    }
+
+    return value.ullVal == NumericValue ? TRUE : FALSE;
+}
+
+extern "C" BOOL DLL_EXPORT STDMETHODCALLTYPE Marshal_ByValue_Single(VARIANT value)
+{
+    if (value.vt != VT_R4)
+    {
+        printf("Invalid format. Expected VT_R4.\n");
+        return FALSE;
+    }
+
+    return value.fltVal == NumericValue ? TRUE : FALSE;
+}
+
+extern "C" BOOL DLL_EXPORT STDMETHODCALLTYPE Marshal_ByValue_Double(VARIANT value)
+{
+    if (value.vt != VT_R8)
+    {
+        printf("Invalid format. Expected VT_R8.\n");
+        return FALSE;
+    }
+
+    return value.dblVal == NumericValue ? TRUE : FALSE;
+}
+
+extern "C" BOOL DLL_EXPORT STDMETHODCALLTYPE Marshal_ByValue_Char(VARIANT value)
+{
+    if (value.vt != VT_UI2)
+    {
+        printf("Invalid format. Expected VT_UI2.\n");
+        return FALSE;
+    }
+
+    return value.uiVal == CharValue ? TRUE : FALSE;
+}
+
+extern "C" BOOL DLL_EXPORT STDMETHODCALLTYPE Marshal_ByValue_String(VARIANT value)
+{
+    static BSTR expected = NULL;
+
+    if (expected == NULL)
```

Comment:
Please don't do this. There are various issues with this pattern. instead `static BSTR expected = SysAllocString(...)`. But in this case, like I said above, don't compute the expected on this side of the test. Have some return for full round trip testing.